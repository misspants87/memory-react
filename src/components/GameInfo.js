import React from 'react';

const GameInfo = props => {

  return (
    <div>
      <button onClick = {props.fillSquares}>Begin Game</button>
        <ul>
          <li>{`Next Player: ${props.playerA.name}`}</li>
          <li>{`${props.playerA.name}'s matches: ${props.playerA.matches}`}</li>
          <li>{`${props.playerB.name}'s matches: ${props.playerB.matches}`}</li>
        </ul>
      </div>
    )
}

export default GameInfo;
