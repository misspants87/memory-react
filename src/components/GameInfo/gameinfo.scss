.game-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
  font-family: 'Roboto';
  font-size: 20pt;
  width: 100%;


  .matches {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 40%;
  }
}
