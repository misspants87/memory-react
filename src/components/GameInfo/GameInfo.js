import React from 'react';
import PropTypes from 'prop-types';
import './gameinfo.scss';

const GameInfo = props => {
  // const beginGame = props.gameSquares.length > 0 ? null : <button onClick = {props.fillSquares}>Begin Game</button>;

// {beginGame}
  return (
    <div className='game-info' data-test="GameInfo Component">

      <h4>{`Next Player: ${props.whichPlayer}`}</h4>

      <div className='matches'>
        <p>{`${props.playerA.name}'s matches: ${props.playerA.matches}`}</p>
        <p>{`${props.playerB.name}'s matches: ${props.playerB.matches}`}</p>
      </div>
    </div>
    )
}

GameInfo.propTypes = {
  playerA: PropTypes.object,
  playerB: PropTypes.object,
  fillSquares: PropTypes.func,
}

export default GameInfo;
