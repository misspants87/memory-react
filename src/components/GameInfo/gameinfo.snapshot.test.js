import React from 'react';
import renderer from 'react-test-renderer';
import GameInfo from './GameInfo';

test('should render GameInfo Component correctly ', () => {
  //render component
  let component = renderer.create(
    <GameInfo
      playerA =  {{name: 'Nick', matches: 2}}
      playerB = {{name: 'Nora', matches: 2}}
      fillSquares= {() => 'fake fn'}
    />
  )

  //turn component into JSON
  let tree = component.toJSON();
  //compare tree to snapshot
  expect(tree).toMatchSnapshot()
})
