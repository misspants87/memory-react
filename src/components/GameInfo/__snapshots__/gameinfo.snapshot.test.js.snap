// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GameInfo Component correctly  1`] = `
<div
  data-set="GameInfo Component"
>
  <div>
    <h4>
      Next Player: Nick
    </h4>
    <p>
      Nick's matches: 2
    </p>
    <p>
      Nora's matches: 2
    </p>
  </div>
</div>
`;
